File: sample-android/src/java/java/root/AndroidApplication.java
package root;

import android.app.Application;

import androidx.annotation.RestrictTo;
import testaccessors.RequiresAccessor;

public final class AndroidApplication extends Application {
  @RequiresAccessor(
      requires = {RequiresAccessor.AccessorType.TYPE_GETTER, RequiresAccessor.AccessorType.TYPE_SETTER},
      androidXRestrictTo = @RestrictTo({RestrictTo.Scope.TESTS, RestrictTo.Scope.LIBRARY}))
  private final String aField = "holahola";
  @RequiresAccessor(
      requires = {RequiresAccessor.AccessorType.TYPE_GETTER, RequiresAccessor.AccessorType.TYPE_SETTER},
      androidXRestrictTo = @RestrictTo(RestrictTo.Scope.TESTS))
  private static Object aStaticField = null;
}

File: sample-android/src/kotlin/java/root/AndroidApplication.kt
package root

import android.app.Application
import testaccessors.RequiresAccessor

@RequiresAccessor(requires = [RequiresAccessor.AccessorType.TYPE_GETTER, RequiresAccessor.AccessorType.TYPE_SETTER])
private val aTopLevelField: String? = null

class AndroidApplication : Application() {
  @RequiresAccessor(requires = [RequiresAccessor.AccessorType.TYPE_GETTER, RequiresAccessor.AccessorType.TYPE_SETTER])
  private val aField: String? = null
  private companion object {
    @RequiresAccessor(requires = [RequiresAccessor.AccessorType.TYPE_GETTER, RequiresAccessor.AccessorType.TYPE_SETTER])
    private var aFieldInAPrivateCompanionObject: Any? = null
  }
}
